Create poke card component

erecoba opened the pull request:
Creation of a poke card with a 3d animation when the cursor is over the card.

erecoba committed:
initial commit

wdonet commented on src/components/PokeCard.js:
```diff
@@ -0,0 +1,189 @@
+import React, { useState, useRef } from "react";
+import { mockPokemonData } from "../mock/pokeData";
```
it is better if you load mock data out of the component, then pass the value as properties

erecoba replied:
Yes, you're right, I already did this, thank you!

App.js
```javascript
import { mockPokemonData } from "./mock/pokeData";

export default function App() {
  return (
    <div className="App">
      <h1>Welcome to your second mini challenge!</h1>
      <PokeCard mockPokemonData={mockPokemonData} />
    </div>
  );
}
```

PokeCard.js
```javascript

function PokeCard({ mockPokemonData }) {// <---- here it comes
  const [cardStyle, setCardStyle] = useState(stylesNormalCard());
  const [spriteStyle, setSpriteStyle] = useState(styleNormalSprite());
  const [textStyle, setTextStyle] = useState(styleNormalText());
  const cardRef = useRef(null);

  ...
}
```

wdonet commented on src/components/PokeCard.js:
```diff
@@ -0,0 +1,189 @@
+  );
+};
+
+const styles = {
```
better to handle styling in a separate file

erecoba replied:
I remove that constant and I used a css file instead, like you suggested 😃, now it looks better.

PokeCard.js
```javascript
import React, { useState, useRef } from "react";
import "./PokeCard.css";

...

const Card = (props) => {
  return (
    <div
      ref={props.parentRef}
      className="PokeCard__card"// <--- A className here
      style={{ ...props.dynamicStyle }}
      onMouseMove={props.onMouseMove}
      onMouseLeave={props.onMouseLeave}
    >
      {props.children}
    </div>
  );
};

const CardTitle = (props) => {
  return (
    <h1 className="PokeCard__text" style={{ ...props.dynamicStyle }}>// <--- A className here
      {props.children}
    </h1>
  );
};

const CardSprites = (props) => {
  return (
    <div>
      {props.sprites.map((sprite, i) => {
        return (
          <img
            key={i}
            src={sprite.img}
            alt={props.alt}
            className="PokeCard__img"// <--- A className here
            style={{ ...props.dynamicStyle }}
          />
        );
      })}
    </div>
  );
};

const CardLinkVideo = (props) => {
  return (
    <p style={{ ...props.dynamicStyle }}>
      <a
        href={props.href}
        target="_blank"
        rel="noopener noreferrer"
        className="PokeCard__text"// <--- A className here
      >
        {props.children}
      </a>
    </p>
  );
};
```

wdonet commented on src/components/PokeCard.js:
```diff
@@ -0,0 +1,189 @@
+        href={props.href}
+        target="_blank"
+        rel="noreferrer"
+        style={styles.card.text}
```
It is better to style components using classnames instead

erecoba replied:
It's done, now I looks better :D

wdonet commented on src/components/PokeCard.js:
```diff
@@ -0,0 +1,189 @@
+    <div
+      ref={props.parentRef}
+      style={{ ...styles.card.container, ...props.dynamicStyle }}
+      onMouseMove={props.onMouseMove}
```
awesome! animation looks good 💯

erecoba replied:
thank you for your comments !!!!

wdonet reviewed (approved):
There's a warning in your code when running:
![image](https://user-images.githubusercontent.com/1031454/105655290-6a2e1b80-5e85-11eb-9134-51b537d8c29f.png)

but Good job! 🤖  👍🏽

erecoba committed:
Apply suggestions
There are a few suggestions about using, classNames and set the poke data into the props from its implementation. This allows me to split specifics actions and task in its own place.
